src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -449,4 +461,51 @@ TEST_F(RetrieveInstancesWidgetTestStarted, ProjectChangeFailed) {
   // "Test Project 2"
 }
 
+TEST_F(RetrieveInstancesWidgetTestStarted, AllCheckboxSuccessful) {
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadInstances(std::optional<Project>(std::nullopt),
+                            InstanceListScope(InstanceListScope::kAllReservedInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<QVector<Instance>>>(kTestInstancesProject1All)));
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadInstances(std::optional<Project>(std::nullopt),
+                            InstanceListScope(InstanceListScope::kOnlyOwnInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<QVector<Instance>>>(kTestInstancesProject1)));
+
+  QTest::mouseClick(all_check_box_, Qt::MouseButton::LeftButton);
+  QCoreApplication::processEvents();
+
+  VerifyLastLoadingReturnedInstanceList(kTestInstancesProject1All);
+  VerifyAndClearSignalsOfSuccessfulLoadingCycle();
+  VerifyAllElementsAreEnabled();
+  EXPECT_TRUE(all_check_box_->isChecked());
+  EXPECT_EQ(LoadInstancesScopeFromPersistentStorage(), InstanceListScope::kAllReservedInstances);
+
+  QTest::mouseClick(all_check_box_, Qt::MouseButton::LeftButton);
+  QCoreApplication::processEvents();
+
+  VerifyLastLoadingReturnedInstanceList(kTestInstancesProject1);
+  VerifyAndClearSignalsOfSuccessfulLoadingCycle();
+  VerifyAllElementsAreEnabled();
+  EXPECT_FALSE(all_check_box_->isChecked());
+  EXPECT_EQ(LoadInstancesScopeFromPersistentStorage(), InstanceListScope::kOnlyOwnInstances);
+}
+
+TEST_F(RetrieveInstancesWidgetTestStarted, AllCheckboxFail) {
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadInstances(std::optional<Project>(std::nullopt),
+                            InstanceListScope(InstanceListScope::kAllReservedInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<QVector<Instance>>>(ErrorMessage{"error"})));
+
+  QTest::mouseClick(all_check_box_, Qt::MouseButton::LeftButton);
+
+  QMetaObject::invokeMethod(
+      &widget_, []() { QCoreApplication::exit(); }, Qt::QueuedConnection);
+  QCoreApplication::exec();
```

Comment:
I'm curious. Why does QCoreApplicaton::processEvents does not work in this case?

Reply:
Since `LoadInstances` returns an error, the is a QMessageBox displayed to the user. If I understand correctly: Since the QMessageBox is modal, a new event queue is started and that means processEvents becomes blocking.

Reply:
Got it. Thanks.